crates/rlp/Cargo.toml

```diff
@@ -33,3 +33,8 @@ derive = ["dep:alloy-rlp-derive"]
 
 arrayvec = ["dep:arrayvec"]
 smol_str = ["dep:smol_str"]
+
+[[example]]
+name = "enum"
+crate-type = ["staticlib"]
```

Comment:
What's this `crate-type` for?

Reply:
It's kinda unnecessary, the difference is without noting the example is a library the error for `cargo run --example enum` is `error[E0601]: `main` function not found in crate `r#enum`

Whereas when the lib demarcation is there, the error is `error: example target `enum` is a library and cannot be executed`.

Unless you have a preference, I can remove it so it is simpler.

Reply:
I'd prefer if we kept the default examples behavior. Can you move the tests to a main function?

Reply:
Actually this is sort of annoying, the compile error shows up with the language server if no `fn main()` is present.

Reply:
@DaniPopes what would the default behavior be? Should I do a roundtrip serialization / deserialization in the main?

Reply:
Examples are binaries by default, and you illustrate it in the main function

Reply:
Heh okay sorry for so much back and forth -- did you want both the tests and an example in main or just the main example?

Reply:
Just the main function, it's fine as is now